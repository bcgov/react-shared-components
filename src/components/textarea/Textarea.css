.bcgov-text-input {
  font-family: "Noto Sans", Verdana, Arial, sans-serif;
  font-size: 18px;
  border: 2px solid #606060;
  margin-top: 5px;
  margin-bottom: 15px;
  border-radius: 4px;
  padding: 5px;
  resize: none;
  max-width: 100%;
}

.bcgov-text-input:focus {
  outline: 4px solid #3b99fc;
  outline-offset: 1px;
}
