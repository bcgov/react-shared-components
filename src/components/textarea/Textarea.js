import React from "react";
import PropTypes from "prop-types";

import "./Textarea.css";

export const Textarea = ({ id, label, onChange }) => {
  return (
    <>
      {label && (
        <div className="text-label">
          <label htmlFor={id}>{label}</label>
        </div>
      )}
      <textarea
        id={id}
        className="text-input"
        rows="8"
        cols="60"
        onChange={(event) => onChange(event.target.value)}
      />
    </>
  );
};

Textarea.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

Textarea.defaultProps = {
  label: "",
};
