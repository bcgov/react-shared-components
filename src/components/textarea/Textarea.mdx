import { Meta, Story } from "@storybook/addon-docs/blocks";
import { Textarea } from "./Textarea";

<Meta title="Textarea" component={Textarea} />

# Textarea

A textarea allows users to input multiple lines of text.

## When to Use

Users need to enter more than a single line of text

## When not to Use

When users need to enter only one line of text, use a text input instead

## API

| Prop       | Type     | Required | Description                                              | Default |
| ---------- | -------- | -------- | -------------------------------------------------------- | ------- |
| label      | string   | false    | value of the heading label for the textarea              | ""      |
| onChange   | function | true     | function describing what to do on text area value change |         |
| id         | string   | true     | value uniquely identifying the textarea                  |         |
| isRequired | boolean  | false    | determines if the textarea is mandatory or not           | false   |

<br />

## With Label

Textarea with a label for an additional description of what text needs to be entered.

<Story name="With Label" id="textarea--with-label"></Story>

```bash
<Textarea id="1" onChange={action("onChange")} label="Can you provide more detail?" />;
```

## Without Label

Textarea without any heading label.

<Story name="Without Label" id="textarea--without-label"></Story>

```bash
<Textarea id="1" onChange={action("onChange")} />;
```
