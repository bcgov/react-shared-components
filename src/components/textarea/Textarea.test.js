import React from "react";
import { render, fireEvent, getByRole } from "@testing-library/react";
import { Textarea } from "./Textarea";
import testBasicSnapshot from "../../TestHelper";

describe("Textarea Component", () => {
  const label = "some label";
  const onChange = jest.fn();

  test("with label matches the snapshot", () => {
    const textArea = <Textarea id="1" label={label} onChange={onChange} />;

    testBasicSnapshot(textArea);
  });

  test("without label matches the snapshot", () => {
    const textArea = <Textarea id="1" onChange={onChange} />;

    testBasicSnapshot(textArea);
  });

  test("on change updates the value in the textarea", () => {
    const { container } = render(
      <Textarea id="1" label={label} onChange={onChange} />
    );

    fireEvent.change(getByRole(container, "textbox"), {
      target: { value: "examplevalue" },
    });

    expect(getByRole(container, "textbox").value).toBe("examplevalue");
  });
});
