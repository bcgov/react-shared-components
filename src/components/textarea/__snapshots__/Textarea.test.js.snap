// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Component when is required matches the snapshot 1`] = `
Array [
  <div
    className="input-label"
  >
    <label
      htmlFor="1"
    >
      some label
    </label>
    <span
      className="mandatory"
      id="asterisk"
    >
      *
    </span>
     
  </div>,
  <textarea
    className="text-input"
    cols="60"
    id="1"
    onChange={[Function]}
    rows="8"
  />,
]
`;

exports[`Textarea Component with label matches the snapshot 1`] = `
Array [
  <div
    className="input-label"
  >
    <label
      htmlFor="1"
    >
      some label
    </label>
    
     
  </div>,
  <textarea
    className="text-input"
    cols="60"
    id="1"
    onChange={[Function]}
    rows="8"
  />,
]
`;

exports[`Textarea Component without label matches the snapshot 1`] = `
Array [
  "",
  <textarea
    className="text-input"
    cols="60"
    id="1"
    onChange={[Function]}
    rows="8"
  />,
]
`;
